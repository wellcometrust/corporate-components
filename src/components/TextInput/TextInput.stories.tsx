import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, text } from '@storybook/addon-knobs';

import FormField from 'FormField';
import FormFieldError from 'FormFieldError';
import Label from 'Label';

import TextInput from './TextInput';

const TextInputExample = () => {
  const isDisabled = boolean('isDisabled', false);
  const isInvalid = boolean('isInvalid', false);
  const isRequired = boolean('isRequired', false);

  return (
    <TextInput
      id="text-input"
      isDisabled={isDisabled}
      isInvalid={isInvalid}
      isRequired={isRequired}
      name="text-input"
      type="text"
    />
  );
};

const TextInputFormFieldExample = () => {
  const label = text('label', 'Input Label');
  const isDisabled = boolean('isDisabled', false);
  const isInvalid = boolean('isInvalid', false);
  const isRequired = boolean('isRequired', false);

  return (
    <FormField>
      <Label htmlFor="text-input" isDisabled={isDisabled} text={label} />
      <TextInput
        id="text-input"
        isDisabled={isDisabled}
        isInvalid={isInvalid}
        isRequired={isRequired}
        name="text-input"
        type="text"
      />
      {isInvalid && (
        <FormFieldError id="text-input-error" errors="This field is invalid" />
      )}
    </FormField>
  );
};

const stories = storiesOf('Components/TextInput', module);

stories.add('Simple', TextInputExample);
stories.add('Form Field (with label)', TextInputFormFieldExample);
