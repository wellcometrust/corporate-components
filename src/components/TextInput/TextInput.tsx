import React, { forwardRef } from 'react';
import cx from 'classnames';

type TextInputProps = {
  className?: string;
  name: string;
  id: string;
  required: boolean;
};

export const TextInput = forwardRef(
  (
    { className, id, name, required }: TextInputProps,
    ref: React.Ref<HTMLInputElement>
  ) => {
    const classNames = cx('cc-text-input', {
      [className]: className
    });

    return (
      <input
        className={classNames}
        id={id}
        name={name}
        ref={ref}
        required={required}
        type="text"
      />
    );
  }
);

export default TextInput;
