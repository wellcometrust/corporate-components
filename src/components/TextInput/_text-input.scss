// ----------------------------------
// UI Components
// TextInput
// ----------------------------------
// Design System 1.0 Alpha
// 2019-11-11
// ----------------------------------

.cc-text-input {
  @extend %text-input-base;
}
