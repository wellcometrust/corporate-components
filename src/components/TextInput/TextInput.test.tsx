import React, { ComponentProps } from 'react';
import { render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';
import TextInput from './TextInput';

const setup = (props: Partial<ComponentProps<typeof TextInput>> = {}) => {
  const defaultProps: ComponentProps<typeof TextInput> = {
    id: 'test',
    name: 'test',
    type: 'text',
    ...props
  };

  return render(<TextInput {...defaultProps} />);
};

describe('<TextInput />', () => {
  it('renders TextInput', () => {
    const { container } = setup({});
    expect(container).toMatchSnapshot();
  });
  it('allows to type ', () => {
    setup({});
    const textbox = screen.getByRole('textbox');
    user.type(textbox, 'abc');
    expect(textbox).toHaveValue('abc');
  });
});
