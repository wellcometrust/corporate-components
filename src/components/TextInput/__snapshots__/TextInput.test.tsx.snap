// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> renders TextInput 1`] = `
<div>
  <input
    class="cc-text-input cc-text-input--text"
    id="test"
    name="test"
    type="text"
  />
</div>
`;
